### ISSUE TITLE
MacOS configure error

### ISSUE THREAD

**Issue by ohhmm** [meta-toolkit/meta#190]
CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  **Failed to find a suitable aligned allocation routine**
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:120 (CompilerKludges)

**Comment by skystrife** [comment]
Can you give me the full CMake output? Also, what version of macOS are you running?

**Comment by ohhmm** [comment]
<img width="486" alt="2017-09-05 09 01 09" src="https://user-images.githubusercontent.com/190779/30047270-e3f7f89c-9218-11e7-8e7f-d5cf32f4b53e.png">


-- No build type selected, defaulting to Release
-- Searching for ICU 58.2
-- ICU include dirs: /opt/local/include
-- ICU libraries: /opt/local/lib/libicudata.dylib;/opt/local/lib/libicui18n.dylib;/opt/local/lib/libicuuc.dylib
-- Using regular malloc; consider installing jemalloc
CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  Failed to find a suitable aligned allocation routine
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:120 (CompilerKludges)


Configuring incomplete, errors occurred!
See also "/Users/sergejkrivonos/Sources/build/meta-xcode/CMakeFiles/CMakeOutput.log".
See also "/Users/sergejkrivonos/Sources/build/meta-xcode/CMakeFiles/CMakeError.log".

[logs.zip](https://github.com/meta-toolkit/meta/files/1276245/logs.zip)

**Comment by skystrife** [comment]
It looks like you're generating an XCode project? What was your exact `cmake` command? Which version of XCode are you running?

**Comment by ohhmm** [comment]
generating for Xcode Version 8.3.3 (8E3004b), using cmake-gui, no extra options.

**Comment by ohhmm** [comment]
command `cmake <path>`
 Xcode Version 8.3.3 (8E3004b)

**Comment by skystrife** [comment]
I think this should be fixed on the `develop` branch. Can you try a

```bash
git checkout develop
git submodule update --init --recursive
```

and have another go?

**Comment by ohhmm** [comment]
CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  Failed to find a suitable aligned allocation routine
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:124 (CompilerKludges)

**Comment by ohhmm** [comment]
I recreated new cache and it works on develop.

**Comment by ohhmm** [comment]
thank you @skystrife

**Comment by pharrukh** [comment]
@skystrife it does not work for me. I get the same error.

`CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  Failed to find a suitable aligned allocation routine
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:124 (CompilerKludges)`

Here are the two files related to the build:
[CMakeError.log](https://github.com/meta-toolkit/meta/files/1404261/CMakeError.log)
[CMakeOutput.log](https://github.com/meta-toolkit/meta/files/1404262/CMakeOutput.log)

Here is the version of my OS: 10.12.6 (16G29)

**Comment by borsik** [comment]
Same issue with OS: 10.13 (17A405)

**Comment by skystrife** [comment]
Can you give me the terminal output of the `cmake` invocation as well? Also, double-check that you are on the `develop` branch (see my comment above on Sep 9).

A workaround for the meantime is to not use the Xcode generator and instead use Makefiles, but I'd obviously like to get the Xcode generator working for you.

**Comment by borsik** [comment]
```# olzhas @ MacBook-Air-Olzhas in ~/university/ir/meta/build on git:develop o [18:54:08]
$ CXX=clang++ cmake ../ -DCMAKE_BUILD_TYPE=Release -DICU_ROOT=/usr/local/opt/icu4c
-- The C compiler identification is AppleClang 9.0.0.9000038
-- The CXX compiler identification is AppleClang 9.0.0.9000038
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Found ZLIB: /usr/lib/libz.dylib (found version "1.2.11")
-- Looking for lzma_auto_decoder in /usr/local/lib/liblzma.dylib
-- Looking for lzma_auto_decoder in /usr/local/lib/liblzma.dylib - found
-- Looking for lzma_easy_encoder in /usr/local/lib/liblzma.dylib
-- Looking for lzma_easy_encoder in /usr/local/lib/liblzma.dylib - found
-- Looking for lzma_lzma_preset in /usr/local/lib/liblzma.dylib
-- Looking for lzma_lzma_preset in /usr/local/lib/liblzma.dylib - found
-- Found LibLZMA: /usr/local/include (found version "5.2.3")
-- Searching for ICU 58.2
-- Found ICU: /usr/local/opt/icu4c/lib/libicudata.dylib;/usr/local/opt/icu4c/lib/libicui18n.dylib;/usr/local/opt/icu4c/lib/libicuuc.dylib (Required is at least version "58.2")
-- ICU version found is 59.1.0, expected 58.2; attempting to build ICU from scratch...
-- ICU include dirs: /Users/olzhas/university/ir/meta/build/deps/icu-58.2/include
-- ICU libraries: icui18n;icuuc;icudata
-- Locating libc++...
-- Located libc++: /usr/lib/libc++.dylib
-- Located libc++ include path: /Library/Developer/CommandLineTools/usr/include/c++/v1/
--     Locating libc++'s abi...
--     Found libc++abi: /usr/lib/libc++abi.dylib
-- Found LIBCXX: /usr/lib/libc++.dylib
-- Using jemalloc: /usr/local/lib/libjemalloc.dylib
-- Performing Test META_HAS_ALIGNED_ALLOC
-- Performing Test META_HAS_ALIGNED_ALLOC - Failed
-- Performing Test META_HAS_POSIX_MEMALIGN
-- Performing Test META_HAS_POSIX_MEMALIGN - Failed
CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  Failed to find a suitable aligned allocation routine
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:124 (CompilerKludges)


-- Configuring incomplete, errors occurred!
See also "/Users/olzhas/university/ir/meta/build/CMakeFiles/CMakeOutput.log".
See also "/Users/olzhas/university/ir/meta/build/CMakeFiles/CMakeError.log".
```

[CMakeOutput.log](https://github.com/meta-toolkit/meta/files/1419298/CMakeOutput.log)
[CMakeError.log](https://github.com/meta-toolkit/meta/files/1419299/CMakeError.log)

**Comment by billlipeng** [comment]
I got a similar error on macOS Version 10.13.1.
[CMakeError.log](https://github.com/meta-toolkit/meta/files/1516385/CMakeError.log)
[CMakeOutput.log](https://github.com/meta-toolkit/meta/files/1516386/CMakeOutput.log)

develop  CXX=clang++ cmake ../ -DCMAKE_BUILD_TYPE=Release -DICU_ROOT=/usr/local/opt/icu4c
make
-- Searching for ICU 58.2
-- ICU version found is 59.1.0, expected 58.2; attempting to build ICU from scratch...
-- ICU include dirs: /Users/zero/Documents/UIUC/course/CS510/research/meta/build/deps/icu-58.2/include
-- ICU libraries: icui18n;icuuc;icudata
-- Locating libc++...
-- Located libc++: /usr/lib/libc++.dylib
-- Located libc++ include path: /Library/Developer/CommandLineTools/usr/include/c++/v1/
--     Locating libc++'s abi...
--     Found libc++abi: /usr/lib/libc++abi.dylib
-- Using jemalloc: /usr/local/lib/libjemalloc.dylib
CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  Failed to find a suitable aligned allocation routine
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:124 (CompilerKludges)


-- Configuring incomplete, errors occurred!
See also "/Users/zero/Documents/UIUC/course/CS510/research/meta/build/CMakeFiles/CMakeOutput.log".
See also "/Users/zero/Documents/UIUC/course/CS510/research/meta/build/CMakeFiles/CMakeError.log".
make: *** No targets specified and no makefile found.  Stop.

**Comment by eric-ahlgren** [comment]
I am experiencing the same issue as above. I am on macOS 10.13.2, attempting to build from the developer branch:

$ CXX=clang++ cmake ../ -DCMAKE_BUILD_TYPE=Release -DICU_ROOT=/usr/local/opt/icu4c
-- The C compiler identification is AppleClang 9.0.0.9000039
-- The CXX compiler identification is AppleClang 9.0.0.9000039
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Found ZLIB: /usr/lib/libz.dylib (found version "1.2.11")
-- Looking for lzma_auto_decoder in /usr/local/lib/liblzma.dylib
-- Looking for lzma_auto_decoder in /usr/local/lib/liblzma.dylib - found
-- Looking for lzma_easy_encoder in /usr/local/lib/liblzma.dylib
-- Looking for lzma_easy_encoder in /usr/local/lib/liblzma.dylib - found
-- Looking for lzma_lzma_preset in /usr/local/lib/liblzma.dylib
-- Looking for lzma_lzma_preset in /usr/local/lib/liblzma.dylib - found
-- Found LibLZMA: /usr/local/include (found version "5.2.3")
-- Searching for ICU 58.2
-- Found ICU: /usr/local/opt/icu4c/lib/libicudata.dylib;/usr/local/opt/icu4c/lib/libicui18n.dylib;/usr/local/opt/icu4c/lib/libicuuc.dylib (Required is at least version "58.2")
-- ICU version found is 59.1.0, expected 58.2; attempting to build ICU from scratch...
-- ICU include dirs: /Users/ericahlgren/meta/build/deps/icu-58.2/include
-- ICU libraries: icui18n;icuuc;icudata
-- Locating libc++...
-- Located libc++: /usr/lib/libc++.dylib
-- Located libc++ include path: /Library/Developer/CommandLineTools/usr/include/c++/v1/
--     Locating libc++'s abi...
--     Found libc++abi: /usr/lib/libc++abi.dylib
-- Found LIBCXX: /usr/lib/libc++.dylib
-- Using jemalloc: /usr/local/lib/libjemalloc.dylib
-- Performing Test META_HAS_ALIGNED_ALLOC
-- Performing Test META_HAS_ALIGNED_ALLOC - Failed
-- Performing Test META_HAS_POSIX_MEMALIGN
-- Performing Test META_HAS_POSIX_MEMALIGN - Failed
CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  Failed to find a suitable aligned allocation routine
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:124 (CompilerKludges)


-- Configuring incomplete, errors occurred!
See also "/Users/ericahlgren/meta/build/CMakeFiles/CMakeOutput.log".
See also "/Users/ericahlgren/meta/build/CMakeFiles/CMakeError.log".

**Comment by Yevgnen** [comment]
Same here:

 $ CXX=clang++ cmake ../ -DCMAKE_BUILD_TYPE=Release -DICU_ROOT=/usr/local/opt/icu4c
-- The C compiler identification is AppleClang 9.0.0.9000039
-- The CXX compiler identification is Clang 5.0.1
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Found ZLIB: /usr/lib/libz.dylib (found version "1.2.8")
-- Looking for lzma_auto_decoder in /usr/local/lib/liblzma.dylib
-- Looking for lzma_auto_decoder in /usr/local/lib/liblzma.dylib - found
-- Looking for lzma_easy_encoder in /usr/local/lib/liblzma.dylib
-- Looking for lzma_easy_encoder in /usr/local/lib/liblzma.dylib - found
-- Looking for lzma_lzma_preset in /usr/local/lib/liblzma.dylib
-- Looking for lzma_lzma_preset in /usr/local/lib/liblzma.dylib - found
-- Could NOT find LibLZMA (missing: LIBLZMA_INCLUDE_DIR)
-- Searching for ICU 58.2
-- Found ICU: /usr/local/opt/icu4c/lib/libicudata.dylib;/usr/local/opt/icu4c/lib/libicui18n.dylib;/usr/local/opt/icu4c/lib/libicuuc.dylib (Required is at least version "58.2")
-- ICU version found is 60.2.0, expected 58.2; attempting to build ICU from scratch...
-- ICU include dirs: /Users/user/Downloads/meta/build/deps/icu-58.2/include
-- ICU libraries: icui18n;icuuc;icudata
-- Locating libc++...
-- Located libc++: /usr/lib/libc++.dylib
-- Located libc++ include path: /Library/Developer/CommandLineTools/usr/include/c++/v1/
--     Locating libc++'s abi...
--     Found libc++abi: /usr/lib/libc++abi.dylib
-- Found LIBCXX: /usr/lib/libc++.dylib
-- Using jemalloc: /usr/local/lib/libjemalloc.dylib
-- Performing Test META_HAS_ALIGNED_ALLOC
-- Performing Test META_HAS_ALIGNED_ALLOC - Failed
-- Performing Test META_HAS_POSIX_MEMALIGN
-- Performing Test META_HAS_POSIX_MEMALIGN - Failed
CMake Error at deps/meta-cmake/kludges/AlignedAlloc.cmake:54 (message):
  Failed to find a suitable aligned allocation routine
Call Stack (most recent call first):
  deps/meta-cmake/CompilerKludges.cmake:18 (include)
  CMakeLists.txt:124 (CompilerKludges)


-- Configuring incomplete, errors occurred!
See also "/Users/user/Downloads/meta/build/CMakeFiles/CMakeOutput.log".
See also "/Users/user/Downloads/meta/build/CMakeFiles/CMakeError.log".

**Comment by slevental** [comment]
The problem is with include path generated by `FindLIBCXX.cmake`, it doesn't find correct installation directory which is shown in `clang++ -v`, instead it finds `/Library/Developer/CommandLineTools/usr/include` and uses it in includes while compiling aligned allocation test code (and other code eventually), you can add original installation path to your clang by adding it as a prefix (not sure why libc++ location /Library/Developer/CommandLineTools/usr/include is not working):
```
cmake -DCMAKE_SYSTEM_PREFIX_PATH=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include;${CMAKE_SYSTEM_PREFIX_PATH} .. && make -j8
```

**Comment by skystrife** [comment]
Fixed in 92fd041.
